File: BE/user-service/src/main/java/com/s206/user/user/dto/request/UserUpdateRequest.java
package com.s206.user.user.dto.request;

import com.s206.user.user.enums.Goal;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserUpdateRequest {

    private String nickname;
    private String phone;
    private Goal goal;
    private BigDecimal height;
    private BigDecimal weight;
    private LocalDate birthday;
    private Boolean notificationEnabled;
}

File: FE/mobile/src/main/java/com/example/diaviseo/network/user/dto/req/UserUpdateRequest.kt
package com.example.diaviseo.network.user.dto.req

data class UserUpdateRequest(
    val nickname: String? = null,
    val phone: String? = null,
    val height: Double? = null,
    val weight: Double? = null,
    val birthday: String? = null, // ISO 8601 형태로: "2000-01-01"
    val notificationEnabled: Boolean? = null,
    val goal: String? = null
)
